packages/block-library/src/post-date/edit.js

```diff
@@ -3,24 +3,67 @@
  */
 import { useEntityProp, useEntityId } from '@wordpress/core-data';
 import { __experimentalGetSettings, dateI18n } from '@wordpress/date';
+import { Popover, DateTimePicker, ToolbarGroup } from '@wordpress/components';
+import { BlockControls } from '@wordpress/block-editor';
 import { __ } from '@wordpress/i18n';
 
-function PostDateDisplay() {
-	const [ date ] = useEntityProp( 'postType', 'post', 'date' );
+function PostDateEditor( { format, setAttributes, isSelected } ) {
+	const [ siteFormat ] = useEntityProp( 'root', 'site', 'date_format' );
+	const [ date, setDate ] = useEntityProp( 'postType', 'post', 'date' );
 	const settings = __experimentalGetSettings();
-
+	const is12Hour = /a(?!\\)/i.test(
+		settings.formats.time
+			.toLowerCase()
+			.replace( /\\\\/g, '' )
+			.split( '' )
+			.reverse()
+			.join( '' )
+	);
 	return date ? (
 		<time dateTime={ dateI18n( 'c', date ) }>
-			{ dateI18n( settings.formats.date, date ) }
+			{ dateI18n( format || siteFormat || settings.formats.date, date ) }
+			{ isSelected && (
+				<Popover>
+					<DateTimePicker
+						currentDate={ date }
+						onChange={ setDate }
+						is12Hour={ is12Hour }
+					/>
+				</Popover>
+			) }
+			<BlockControls>
+				<ToolbarGroup
```

Comment:
I don't think the format is changed often enough to warrant a placement in the toolbar. It could be just an inspector control.

Reply:
cc @shaunandrews